### PR TITLE
verify signature and get custom ApiConfig

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -16,6 +19,11 @@ const TypeError = "error"
 const Version = "0.0.2.1"
 const ExeFileName = "speedsnitch"
 
+const ConfigPath = "/boot/AppConfig"
+const ConfigFileName = "speedsnitch.txt"
+
+const GPGKeyFileName = "gpg.pubkey"
```
Comment (fillup) [inline agent.go]: lets rename this to speedsnitch.pub so its more easily identifiable on the filesystem

File: agent.go [hunk 2]
```diff
@@ -16,6 +19,11 @@ const TypeError = "error"
 const Version = "0.0.2.1"
 const ExeFileName = "speedsnitch"
 
+const ConfigPath = "/boot/AppConfig"
```
Comment (fillup) [inline agent.go]: This path really only applies to RaspberryPi installs, so we may need to make this more platform dynamic or something

File: agent.go [hunk 3]
```diff
@@ -154,3 +166,71 @@ func GetTaskLogEntry(entryType string) TaskLogEntry {
...
+
+	scanner.Scan()
+	apiConfig.APIKey = scanner.Text()
+
+	return apiConfig
+}
+
+
+// VerifyFileSignature only checks the signature of the target file if there is a gpg key to use
+func VerifyFileSignature(directory, targetFile, signedFile string) error {
```
Comment (fillup) [inline agent.go]: Lets move the public key into the source but support an array of public keys so we can support rotation of keys through built in auto-update feature without having to have end users put a new public key file into place.

File: agent.go (hunk 1 above)
Comment (Baggerone) [reply agent.go]: No longer relevant, since a file won't be involved.

File: agent.go (hunk 3 above)
Comment (Baggerone) [reply agent.go]: Done

File: agent.go (hunk 2 above)
Comment (Baggerone) [reply agent.go]: made it check three different locations to find the AppConfig

Comment (fillup) [reply agent.go]: so you can remove this line now right?

File: agent.go (hunk 1 above)
Comment (fillup) [reply agent.go]: ok, so you can remove this now right?

Comment (Baggerone) [reply agent.go]: Removed

File: agent.go (hunk 2 above)
Comment (Baggerone) [reply agent.go]: Removed

**Pull request merged**